### PR TITLE
[TA3968] feat(stats): add rep/tgt status in stats

### PR DESCRIPTION
**Pull request by utkarshmani1997**
This commit adds a new feature to show replica and controller's
status in iostats.

Signed-off-by: Utkarsh Mani Tripathi <utkarsh.tripathi@mayadata.io>

### CODE REVIEW

**Review by mynktl — COMMENTED** [review]
Changes are good. But why do we need replica status/info for io stats?

File: src/istgt_lu_ctl.c [hunk 1]
```diff
@@ -3379,12 +3380,31 @@ istgt_uctl_cmd_iostats(UCTL_Ptr uctl)
...
+		    json_object_new_int(spec->healthy_rcount +
+			spec->degraded_rcount));
 		json_object_object_add(jobj, "RevisionCounter",
 		    json_object_new_uint64(spec->io_seq));
+		json_object_object_add(jobj, "Status",
+		    json_object_new_string(((spec->ready == 1) ?
+			"RW" : "RO")));
+
+		json_object *jobj_arr = json_object_new_array();
+		TAILQ_FOREACH(replica, &spec->rq, r_next) {
```
Comment (vishnuitta) [inline src/istgt_lu_ctl.c:3393]: accessing spec->rq needs spec->rq_mtx

Comment (utkarshmani1997) [reply src/istgt_lu_ctl.c:3393]: Thanks @vishnuitta for pointing it out.

**Comment by utkarshmani1997** [comment]
@mynktl as it is being consumed by exporter, we need this info.

**Comment by mynktl** [comment]
ok @utkarshmani1997

**Review by mynktl — APPROVED** [review]

**Review by vishnuitta — APPROVED** [review]
changes are good

**Pull request merged**